File: pact-jvm-provider-junit/src/main/java/au/com/dius/pact/provider/junit/MessagePactRunner.java
package au.com.dius.pact.provider.junit;

import au.com.dius.pact.model.Pact;
import au.com.dius.pact.model.v3.messaging.MessagePact;
import org.apache.commons.collections4.ListUtils;
import org.apache.commons.collections4.Predicate;
import org.junit.runners.model.InitializationError;

import java.util.List;

public class MessagePactRunner extends PactRunner {
    public MessagePactRunner(final Class<?> clazz) throws InitializationError {
        super(clazz);
    }

    @Override
    protected List<Pact> filterPacts(List<Pact> pacts) {
        return ListUtils.select(pacts, new Predicate<Pact>() {
            @Override
            public boolean evaluate(Pact pact) {
                return pact.getClass() == MessagePact.class;
            }
        });
    }
}

File: pact-jvm-provider-junit/src/main/kotlin/au/com/dius/pact/provider/junit/MessagePactRunner.kt
package au.com.dius.pact.provider.junit

import au.com.dius.pact.core.model.FilteredPact
import au.com.dius.pact.core.model.Interaction
import au.com.dius.pact.core.model.Pact
import au.com.dius.pact.core.model.messaging.MessagePact

/**
 * Pact runner that only verifies message pacts
 */
open class MessagePactRunner<I>(clazz: Class<*>) : PactRunner<I>(clazz) where I: Interaction {
  override fun filterPacts(pacts: List<Pact<I>>): List<Pact<I>> {
    return super.filterPacts(pacts).filter { pact ->
      pact is MessagePact || (pact is FilteredPact && pact.pact is MessagePact)
    }
  }
}
